### DISCUSSION TITLE
Is there any way to add automatically a non-existing network to the metamask wallet (switch to it as well) ?

### DISCUSSION THREAD

**Discussion by pwiniarski1991** [wevm/wagmi#1729] [Q&A]
I have looked through all questions asked on this board and only found this one: https://github.com/wagmi-dev/wagmi/discussions/1670#discussioncomment-4690853

Based on the answer from @tmm it depends on Connector that is used if it atomatically request to add non-existing network to wallet if user is not using allowed configured list of networks set in the app.

Is there anyway to make it possible for metamask wallet to add non-existing network to wallet if user is using unsupported by app network ?.

If not, then is it possible to catch/listen to metamask rpc error ?. I have only seen this way: https://docs.metamask.io/guide/ethereum-provider.html#events, but there is no error callback then can be used or maybe there is method that return promise with data or error.

**Comment by tmm** [comment]
[`useSwitchNetwork`](https://wagmi.sh/react/hooks/useSwitchNetwork) will add the Chain if users wallet does not have it already.

**Reply by pwiniarski1991** [reply]
@tmm Thank you so much for reply.  I think my issue is more related that network config for Arbitrum Goerli Testnet provided by wagmi/config package is broken because when I provided it manually it started to work as supposed ie. if add network fail then there should be so message/notification displayed and also possiblity to connect to listen on such errors.

I agree with useSwitchNetwork is great and works, but when it is not used at all (explicitly) then how it works ie. I am using just WagmiConfig and connectkit (that shows modal to connect to wallet). Am I missing some details ?
